server/src/main/java/org/opensearch/plugins/PluginInfo.java

```diff
@@ -486,7 +486,7 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         {
             builder.field("name", name);
             builder.field("version", version);
-            builder.field("opensearch_version", opensearchVersionRanges);
+            builder.field("opensearch_version", getOpenSearchVersionRangesString());
```

Comment:
```
    public String getOpenSearchVersionRangesString() {
        if (opensearchVersionRanges == null || opensearchVersionRanges.isEmpty()) {
            return "";
        }
        if (opensearchVersionRanges.size() == 1) {
            return opensearchVersionRanges.get(0).toString();
        }
        return opensearchVersionRanges.stream().map(Object::toString).collect(Collectors.joining(",", "[", "]"));
    }
```
This method can return list of opensearch versions in future. How is that controlled?

Reply:
Also when will we ever return "" from here. That should never happen right?

Reply:
PluginInfo enforces the range size to be exactly one - https://github.com/opensearch-project/OpenSearch/blob/ed67d257938ffae45168ac28e68ef32138b6653a/server/src/main/java/org/opensearch/plugins/PluginInfo.java#L307.

https://github.com/opensearch-project/OpenSearch/blob/ed67d257938ffae45168ac28e68ef32138b6653a/server/src/main/java/org/opensearch/plugins/PluginInfo.java#L143

Hence empty or multiple ranges cannot exist.

Reply:
then should empty range be IllegalStateException. For my understanding is the multi range handling here for future.

Reply:
Right, multi range handling is for future. 
Updated the check to throw IllegalStateException